GSA/digitalgov.gov
# Display upcoming events to community page

## Summary
Update front matter fields and template to display future events on a communities page. 

1. Individual community page will show upcoming events
2. Events tagged with multiple communities will display for each respective community

## Preview

[Link to Preview]()



## Solution

- ef1137df026e184a5b58f1486277d7a9c2039ab1 fixes a variable naming mis-match. Event pages were using `cop_events` and `get-upcomingevents.html` referenced `events_cop`. 

- b8c9ad1f20115f0468b536f73a3312f33c4ca24d removes old `cop_events` field.

- 694ea030b537d17f26bc6dd27ff281345264ea81 enables displaying one event across multiple communities.

**Front Matter Updates**

**event page**

```
---
event_cop:
  - multilingual
  - communicators
---
```

**community page**

```
event_cop: communicators
```



<img width="727" alt="Screenshot 2024-10-17 at 10 07 32 AM" src="https://github.com/user-attachments/assets/a780f68e-71f9-4e4d-80af-fa224f3d45d3">


## How To Test

1. Future date an event if one does not exist

```
date: 2024-10-?? 14:00:00 -0500
end_date: 2024-10-?? 15:00:00 -0500
```

2. Add `event_cop` field to future event using value from `themes/digital.gov/static/workflow/config.yml`
3. Visit community page to see list of upcoming events

## What To Test

1. Display future events for a community page
2. Display no future events for a non tagged community page
3. Display no future events when there are none
4. Display multiple future events for a community page
